src/dagmc.cpp

```diff
@@ -69,35 +159,80 @@ void load_dagmc_geometry()
       model::universes[it->second]->cells_.push_back(i);
     }
 
+    // check for temperature assignment
+    std::string temp_value;
+    if (model::DAG->has_prop(vol_handle, "temp")) {
+      rval = model::DAG->prop_value(vol_handle, "temp", temp_value);
+      MB_CHK_ERR_CONT(rval);
+      double temp = std::stod(temp_value);
+      c->sqrtkT_.push_back(std::sqrt(K_BOLTZMANN * temp));
+    } else {
+      c->sqrtkT_.push_back(std::sqrt(K_BOLTZMANN * settings::temperature_default));
+    }
+
+    // MATERIALS
+
     if (model::DAG->is_implicit_complement(vol_handle)) {
-      // assuming implicit complement is void for now
-      c->material_.push_back(MATERIAL_VOID);
+      if (model::DAG->has_prop(vol_handle, "mat")) {
+        // if the implicit complement has been assigned a material, use it
+        std::string comp_mat = DMD.volume_material_property_data_eh[vol_handle];
+        // Note: material numbers are set by UWUW
+        int mat_number = uwuw.material_library[comp_mat].metadata["mat_number"].asInt();
+        c->material_.push_back(mat_number);
+      } else {
+        // if no material is found, the implicit complement is void
+        c->material_.push_back(MATERIAL_VOID);
+      }
       continue;
     }
 
-    if (model::DAG->has_prop(vol_handle, "mat")){
-      std::string mat_value;
+    // determine volume material assignment
+    std::string mat_value;
+    if (model::DAG->has_prop(vol_handle, "mat")) {
       rval = model::DAG->prop_value(vol_handle, "mat", mat_value);
       MB_CHK_ERR_CONT(rval);
-      to_lower(mat_value);
-
-      if (mat_value == "void" || mat_value == "vacuum") {
-        c->material_.push_back(MATERIAL_VOID);
-      } else {
-        c->material_.push_back(std::stoi(mat_value));
-      }
     } else {
       std::stringstream err_msg;
       err_msg << "Volume " << c->id_ << " has no material assignment.";
       fatal_error(err_msg.str());
     }
+
+    std::string cmp_str = mat_value;
+    to_lower(cmp_str);
+    // material void checks
+    if (cmp_str.find("void") != std::string::npos   ||
+        cmp_str.find("vacuum") != std::string::npos ||
+        cmp_str.find("graveyard") != std::string::npos) {
+      c->material_.push_back(MATERIAL_VOID);
```

Comment:
Just a question here -- in OpenMC we don't have the concept of a purely absorbing cell (like a 0 importance cell in MCNP). Making a material void just means there's nothing there and neutrons will stream through it since it has zero cross section. Is this graveyard concept really the same thing?

Reply:
Nearly? Right now there's an expectation that the model will include a graveyard volume which encompasses all other volumes. In the case of OpenMC, my plan is that the graveyard should have vacuum boundary conditions and a void material. Is that sufficient to leave results unaffected by it's presence?

(I'll make sure to include all of this in the model prep documentation in DAGMC.)

Reply:
Yup, results would be unchanged by such a volume.